nncf/common/pruning/model_analysis.py

```diff
@@ -11,15 +11,132 @@
  limitations under the License.
 """
 
-from typing import Callable, List
+import numpy as np
+
+from typing import Callable, List, Dict
 
 from nncf.common.graph import NNCFGraph
 from nncf.common.graph import NNCFNode
+from nncf.common.tensor import NNCFTensor
+from nncf.common.tensor import NNCFBaseTensorProcessor
 from nncf.common.pruning.clusterization import Cluster
 from nncf.common.pruning.clusterization import Clusterization
 from nncf.common.pruning.operations import BasePruningOp
+from nncf.common.pruning.mask_propagation import MaskPropagationAlgorithm
+from nncf.common.pruning.utils import get_input_masks
+from nncf.common.pruning.utils import is_grouped_conv
 from nncf.common.pruning.utils import find_next_nodes_not_of_types
 from nncf.common.pruning.utils import PruningOperationsMetatypeRegistry
+from nncf.common.pruning.utils import PruningAnalysisDecision
+from nncf.common.pruning.utils import PruningAnalysisReason
+
+
+class SymbolicMaskProcessor(NNCFBaseTensorProcessor):
+    @classmethod
+    def concatenate(cls, tensors: List[NNCFTensor], axis: int) -> NNCFTensor:
+        ret_tensor = np.concatenate([t.tensor for t in tensors], axis=axis)
+        producers = []
+        for tensor in tensors:
+            if tensor.mask_producers is not None:
+                producers.extend(tensor.mask_producers)
+        if not producers:
+            producers = None
+
+        return SymbolicMask(ret_tensor, producers)
+
+    @classmethod
+    def ones(cls, shape: List[int], device) -> NNCFTensor:
+        ret_tensor = np.ones(shape)
+        return SymbolicMask(ret_tensor)
+
+    @classmethod
+    def check_all_close(cls, tensors: List[NNCFTensor]) -> None:
+        for input_mask in tensors[1:]:
+            assert input_mask.tensor.shape == tensors[0].tensor.shape
```

Comment:
I'd confuse this function with `torch.all_close` that check data instead of shape. Maybe `check_all_shapes_equal`?

Reply:
I've added description, please check